src/utils/misc.ts

```diff
@@ -148,8 +150,23 @@ export function saveTokenAllocation(
   entity.timestamp = timestamp;
   entity.recipient = recipient;
   entity.txHash = txHash;
+  entity.tokenDistroAddress = tokenDistroAddress;
   entity.save();
   tokenAllocationIds.push(entityId);
   transactionTokenAllocations.tokenAllocationIds = tokenAllocationIds;
   transactionTokenAllocations.save();
 }
+
+export function getTokenDistroBalance(
+  tokenAddress: string,
+  userAddress: string,
+): TokenDistroBalance {
+  const id = tokenAddress + '-' + userAddress;
+  let tokenDistroBalance = TokenDistroBalance.load(id);
+
+  if (!tokenDistroBalance) {
+    tokenDistroBalance = new TokenDistroBalance(id);
```

Comment:
the entity is not initialized with user address and tokenDistro address

Reply:
constructor says it initializes with the id which is the string joining tokenDistro and userAddress, or am I wrong?

Reply:
Look at other examples, like tokenBalance.

Reply:
Got it I see it now